### PR TITLE
More Sphinx / Doxygen updates

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

- Use 'warning' admonition in Sphinx docs so these will be formatted similar to 'deprecated' admonitions now that we're using the PyData Sphinx theme.
- Add BibTeX entries for more references
- Remove references to obsolete member of `VPStandardStateTP`
- Fix Contributor Covenant link in `README.rst`
- Expand trees in Doxygen by default -- saves having to expand the "Cantera" namespace every time you go to the class / namespace list
- Adjust ordering of Doxygen pages to move auxiliary pages towards the bottom, and emphasize "class" documentation over the "namespace" documentation
- Make transport properties a top-level group in Doxygen, and move Phase and ThermoPhase into base `thermoprops` group

**If applicable, provide an example illustrating new features this pull request is introducing**

Member function with `warning` and `versionadded` directives:
<hr />
image
<hr />

Default view of the "Classes" page:
<hr />

image

<hr />

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: doc/doxygen/cantera.bib [hunk 1]
```diff
@@ -214,6 +279,16 @@ @article{niemeyer2017
     url = {https://dx.doi.org/10.1016/j.cpc.2017.02.004},
     volume = {21},
     year = {2017}}
+@article{pedersen1993,
+    author = {T.~Pedersenand R.~.C.~Brown},
```
Comment (ischoegl) [inline doc/doxygen/cantera.bib]:
```suggestion
@article{pederson1993,
    author = {T.~Pederson and R.~.C.~Brown},
```
Elsewhere it's "Pederson"? ...

File: doc/doxygen/cantera.bib [hunk 2]
```diff
@@ -226,6 +301,17 @@ @article{perini2012
     url = {https://dx.doi.org/10.1021/ef300747n},
     volume = {26},
     year = {2012}}
+@article{pitzer1975,
+    author = {Pitzer, Kenneth S.},
```
Comment (ischoegl) [inline doc/doxygen/cantera.bib]:
```suggestion
    author = {K.~S.~Pitzer},
```

File: doc/doxygen/cantera.bib [hunk 3]
```diff
@@ -280,6 +366,15 @@ @article{sengers1986
...
     }
+@article{silvester1977,
+    author = {L.~F.~Silvester and K.~S.~Pitzer},
+    title = {Thermodynamics of electrolytes. 8. High-temperature properties, including
+        enthalpy and heat capacity, with application to sodium chloride},
+    journal = {Journal of Physical Chemistry},
+    volume = {81},
+    number = {19},
+    pages = {1822--1828},
+    year = {1977}}
```
Comment (ischoegl) [inline doc/doxygen/cantera.bib:378]:
```suggestion
    url = {https://doi.org/10.1021/j100534a007},
    year = {1977}}
```

File: doc/doxygen/cantera.bib [hunk 4]
```diff
@@ -82,6 +112,31 @@ @article{han2015
...
+    journal = {Geochimica et Cosmochimica Acta},
+    volume = {44},
+    number = {7},
+    month = jul,
+    url = {https://doi.org/10.1016/0016-7037(80)90287-2},
+    doi = {10.1016/0016-7037(80)90287-2},
+    pages = {981--997},
+    year = {1980}}
+@article{johnson1992,
+    author = {Johnson, James W. and Oelkers, Eric H. and Helgeson, Harold C.},
```
Comment (ischoegl) [inline doc/doxygen/cantera.bib]:
```suggestion
    author = {J.~W.~Johnson and E.~H.~Oelkers and H.~C.~Helgeson},
```

**Review by ischoegl — CHANGES_REQUESTED** [review]
Thanks, @speth! A couple of BibTeX quibbles (doxygen uses full first/middle names if they are provided, resulting in inconsistent entries)

Regarding doxygen, there's also the `@tparam` issue (see [here](https://www.doxygen.nl/manual/commands.html#cmdtparam)) that we may want to address eventually (not necessarily now). It's probably best an enhancement request, but I keep forgetting about it.

File: doc/doxygen/cantera.bib (hunk 1 above)
Comment (speth) [reply doc/doxygen/cantera.bib]: Ah, thanks. It's definitely "Pedersen", if you follow the DOI link. I corrected the incorrect spelling in `IonFlow.h`.

**Comment by speth** [comment]
Thanks for catching those inconsistencies, @ischoegl. This should be good to go now.

**Review by ischoegl — APPROVED** [review]

**Pull request merged**